### ISSUE TITLE
Full Attributes for HTMLConverter.convert

### ISSUE THREAD

**Issue by andyl** [elixir-tools/tableau#99]
I'm experimenting with MDEx 2.0 & expanding liquid tags in the markdown.  I can't figure out how to expose the data/site assigns to the `HTMLConverter.convert` function.

Firstly I wrote a converter module to handle liquid tags - this seems to work fine:
```elixir
defmodule Tableau.MdexConvert do

  @moduledoc false

  def liquid_tags(markdown, assigns) do

    str_assigns = assigns |> stringify_keys()

    markdown
    |> MDEx.parse_document!()
    |> MDEx.traverse_and_update(fn
      # render each text as liquid template
      {node, attrs, children} ->
        children =
          Enum.reduce(children, [], fn
            child, acc when is_binary(child) ->
              with {:ok, template} <- Solid.parse(child),
                   {:ok, rendered} <- Solid.render(template, str_assigns) do
                [to_string(rendered) | acc]
              else
                _ -> [child | acc]
              end

            child, acc ->
              [child | acc]
          end)
          |> Enum.reverse()

        {node, attrs, children}
    end)
  end

  defp stringify_keys(map) do
    Map.new(map, fn {k, v} -> {Atom.to_string(k), v} end)
  end

end
```

Secondly I modified the `HTMLConvert` modules to handle liquid tags.  This works fine with one problem.  The attrs contain the markdown frontmatter variables, but not the site's `data` or `site` assigns.  

```elixir
defmodule Tableau.PageExtension.Pages.HTMLConverter do
  @moduledoc false
  def convert(_filepath, body, attrs, _opts) do

    {:ok, config} = Tableau.Config.new(Map.new(Application.get_env(:tableau, :config, %{})))

    body
    |> Tableau.MdexConvert.liquid_tags(attrs)
    |> MDEx.to_html!(config.markdown[:mdex])
  end
end
```

@mhanberg can you point out how to expose the site/data assigns in the `HTMLConvert` modules?  TIA

**Comment by mhanberg** [comment]
Unfortunately I'm messing with this part of the codebase as we speak, but fortunately it should make this easier to achieve I believe

Also, nitpick, its MDEx v0.2.0, not v2.0 😂

**Comment by mhanberg** [comment]
Also for posterity, the "attrs" for these functions are the yaml front matter. Its poorly named and part of what I'm changing at the moment.

**Comment by andyl** [comment]
Thanks for the note I'll wait for your change.  LOL and thanks for the fix re: v0.2.

**Comment by andyl** [comment]
NICE!!
